### PR TITLE
add check for unique device ids 

### LINKED ISSUE #8634: Error against specifying duplicate GPU device ids?
### Discussed in https://github.com/PyTorchLightning/pytorch-lightning/discussions/8630

> I have a single GPU, but I would like to spawn multiple replicas on that single GPU and train a model with DDP. Of course, each replica would have to use a smaller batch size in order to fit in memory. (For my use case, I am not interested in having a single replica with a large batch size). I tried to pass `--gpus "0,0"` to the Lightning Trainer, and it managed to spawn two processes on the same GPU:
```
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
initializing ddp: GLOBAL_RANK: 0, MEMBER: 1/2
initializing ddp: GLOBAL_RANK: 1, MEMBER: 2/2
```
> But in the end it crashed with `RuntimeError: NCCL error in: /pytorch/torch/lib/c10d/ProcessGroupNCCL.cpp:911, invalid usage`.

Given that this isn't supported by the underlying backends, we can explicitly check against duplicate device ids and error out in the trainer if so: https://github.com/PyTorchLightning/pytorch-lightning/blob/c99e2fe0d2bf713f35054eaa0d521ee7f6030786/pytorch_lightning/utilities/device_parser.py#L53-L91

### PR DESCRIPTION
**Pull request by nasnoisaac**
## What does this PR do?

Fixes #8634 

Check the uniqueness of  gpu ids. The backend doesn't support duplicate GPU.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/device_parser.py [hunk 1]
```diff
@@ -188,6 +192,21 @@ def _get_all_available_gpus() -> List[int]:
...
+def _check_unique(device_ids: List[int]) -> None:
+    """
+    Checks that the device_ids are unique.
+
+    Args:
+        device_ids: list of ints corresponding to gpus/tpu_cores indices
+
+    Raises:
+        MisconfigurationException:
+            If ``device_ids`` of GPU/TPUs aren't unique
```
Comment (carmocca) [inline pytorch_lightning/utilities/device_parser.py:204]: Does TPUs need this?

cc @kaushikb11

**Review by carmocca — APPROVED** [review]
Remember to update the CHANGELOG :)

File: pytorch_lightning/utilities/device_parser.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/utilities/device_parser.py:204]: No, it doesn't. As TPUs support training on either 1 or 8 cores or on a single core with a particular index such as `[1]`. We handle such parsing logic internally.

Comment (kaushikb11) [inline pytorch_lightning/utilities/device_parser.py:204]:
```suggestion
            If ``device_ids`` of GPUs aren't unique
```

File: pytorch_lightning/utilities/device_parser.py [hunk 2]
```diff
@@ -188,6 +192,21 @@ def _get_all_available_gpus() -> List[int]:
     return list(range(torch.cuda.device_count()))
 
 
+def _check_unique(device_ids: List[int]) -> None:
+    """
+    Checks that the device_ids are unique.
+
+    Args:
+        device_ids: list of ints corresponding to gpus/tpu_cores indices
```
Comment (kaushikb11) [inline pytorch_lightning/utilities/device_parser.py:200]:
```suggestion
        device_ids: list of ints corresponding to GPUs indices
```

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/utilities/device_parser.py (hunk 1 above)
Comment (nasnoisaac) [reply pytorch_lightning/utilities/device_parser.py]: Thanks for your review. I have updated it.

**Review by kaushikb11 — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added `state_id` property to the `Callback` base class ([#6886](https://github.com/PyTorchLightning/pytorch-lightning/pull/6886))
 
 
--
+- Add check for unique GPU ids ([#8666](https://github.com/PyTorchLightning/pytorch-lightning/pull/8666))
```
Comment (kaushikb11) [inline CHANGELOG.md:15]:
```suggestion
- Added check for unique GPU ids ([#8666](https://github.com/PyTorchLightning/pytorch-lightning/pull/8666))
```

**Review by ananthsub — APPROVED** [review]
thanks for adding this check @nasnoisaac !

**Pull request merged**